Dockerfile

```diff
@@ -0,0 +1,18 @@
+FROM python:3-alpine
+
+ENV CUPS_HOST "127.0.0.1"
+ENV CUPS_PORT "631"
+ENV CUPS_USER "default"
+ENV LISTEN_PORT "9329"
+
+ADD . /opt/cups-exporter
+
+WORKDIR /opt/cups-exporter
+
+EXPOSE 9329
+
+RUN set -xe;\
+  apk add gcc libc-dev cups-dev;\
+  pip install prometheus-client~=0.7.1 pycups~=1.9.74

```

Comment:
I'm not really happy with pinning the dependencies in the dockerfile as well.

On the other hand, I'm not sure whether you would want to have pipenv installed in the container.

Reply:
This is exactly the point, avoid installing **pipenv** in the docker image to make it smaller.

Reply:
You can generate a **requirement.txt** from pipenv and add it to git so it can be used on docker (see https://pypi.org/project/pipenv-to-requirements/ )